help and docs: Security & account management.

alya opened the pull request:
## Notes

- The user-uploaded content section needs some kind of organization pass (e.g., perhaps some of it should live in the help center or on /security), but I'm not sure how best to manage it, and it can be done as a follow-up.
- Please give /bots-overview an especially careful read, as I'm not sure I have the details/framing quite right.

## Follow-ups
- Move security-model.md to securing-your-zulip-server.md.
- Improving organization of the authentication methods doc
- Expanding the doc on protecting your account
- Improving organization of /security


## Screenshots

<details>
<summary>
Change your password (1st commit)
</summary>

[Change your password](https://chat.zulip.org/help/change-your-password)
<img width="1452" height="1300" alt="Screenshot 2025-09-16 at 23 51 53@2x" src="https://github.com/user-attachments/assets/490d9023-9023-4ff0-84d0-854dad2116ef" />

</details>

<details>
<summary>
Protect your account (new) (2nd commit)
</summary>

<img width="2034" height="1584" alt="Screenshot 2025-09-16 at 23 50 40@2x" src="https://github.com/user-attachments/assets/7ce53852-1203-49f0-8a41-ccbe8f9abd05" />

</details>

<details>
<summary>
Logging out (3rd commit)
</summary>

[Logging out](https://chat.zulip.org/help/logging-out)
<img width="1442" height="1662" alt="Screenshot 2025-09-17 at 08 09 08@2x" src="https://github.com/user-attachments/assets/4c503ab1-35cd-4f8f-83b5-50e85139c9d8" />

</details>

<details>
<summary>
Direct messages (5th commit)
</summary>

[Direct messages](https://chat.zulip.org/help/direct-messages)
<img width="1430" height="588" alt="Screenshot 2025-09-17 at 08 21 38@2x" src="https://github.com/user-attachments/assets/f65a687a-5564-4526-8f10-de10f6ea6791" />

</details>

<details>
<summary>
Bots overview (6th commit)
</summary>

[Bots overview](https://chat.zulip.org/help/bots-overview)
<img width="1364" height="1546" alt="Screenshot 2025-09-17 at 09 03 06@2x" src="https://github.com/user-attachments/assets/45c078ca-d434-4135-b43b-2a48832aeca6" />

</details>

<details>
<summary>
Security model (7th commit)
</summary>

[Security model](https://zulip.readthedocs.io/en/latest/production/security-model.html)
<img width="2092" height="1650" alt="Screenshot 2025-09-17 at 10 07 34@2x" src="https://github.com/user-attachments/assets/4dbecb4c-ab4c-4e8f-acb4-edf572cd9887" />
<img width="1512" height="1200" alt="Screenshot 2025-09-17 at 10 07 43@2x" src="https://github.com/user-attachments/assets/cbfd4735-8f50-4ea7-b202-efa66a841eba" />

(+ 2 long sections)

</details>

alexmv commented on starlight_help/src/content/docs/bots-overview.mdx:
```diff
@@ -42,31 +42,32 @@ The **bot type** determines what the bot can do.
 | Outgoing webhook | Generic bot that also receives new messages via HTTP post requests | Third party integrations, most custom bots                      |
 
+A **generic** bot is like a normal Zulip user account that cannot log in via a
+browser. There's a handful of actions no bots can't take, including creating
```
"actions no bots can't take"

timabbott replied:
I don't love this description. I'd say "A generic bot acts like a normal Zulip user that can only access Zulip via the API." seems cleaner

alexmv commented on starlight_help/src/content/docs/bots-overview.mdx:
```diff
@@ -75,6 +76,20 @@ Zulip organization, but administrators can
+permission to send messages as users in your organization. Bots with the
+`can_forge_sender` permission can also see the names of all channels,
+including private channels. This is important for implementing integrations
+like the Jabber, IRC, and Zephyr mirrors.
```
Not sure we should bother to mention Zephyr anymore.

alexmv commented on docs/production/security-model.md:
```diff
@@ -1,207 +1,57 @@
-are rejected, and strong passwords encouraged. The minimum password
-strength allowed is controlled by two settings in
-`/etc/zulip/settings.py`:
+# Securing your Zulip server
```
The filename of this should change, because it's no longer the security model, it's security instructions.

alya replied:
Added a comment to the PR description that we should do this follow-up.

timabbott commented on starlight_help/src/content/docs/change-your-password.mdx:
```diff
@@ -20,6 +20,6 @@ account.
 <FlattenedSteps>
```
This isn't new, but I feel like "You can reset your password ..."  in the copy at the top of the file is not using standard terminology. Typically, "reset" refers to when you are resetting via email, and "change your password" when you do it in the UI.

alya replied:
Is there anything in particular you recommend changing on this page? I think it's ok to use "reset" or "change" when talking about both flows together.

alya replied:
Oh, I think this comment is just mipslaced. Fixing elsewhere.

timabbott commented on starlight_help/src/content/docs/protect-your-account.mdx:
```diff
@@ -0,0 +1,50 @@
+    it](/help/change-your-password#if-youve-forgotten-or-never-had-a-password)
+    using your email account.
+</ZulipTip>
+
```
I think we want another tip here suggesting that you should only use your API key for integrations that need to act as your account; most integrations should use a bot user, which doesn't expose your account and organization to as much risk if the bot's API key is stolen.

timabbott commented on docs/overview/architecture-overview.md:
```diff
@@ -52,9 +52,12 @@ chamber with its own users, channels, customizations, and so on. This
-Center](https://zulip.com/help/).
+more on security considerations and options, see our [guide on securing
+your Zulip server](../production/security-model.md) and the [Zulip help
+center](https://zulip.com/help/).
```
Should this link to zulip.com/security here too?

alya replied:
Oops, I half-finished that change.

timabbott commented:
lgtm aside from the things noted above, most of which we should just handle as follow-ups. (The #general conversation will be easier with this in `main`, etc.)

alya commented on starlight_help/src/content/docs/bots-overview.mdx:
```diff
@@ -42,31 +42,32 @@ The **bot type** determines what the bot can do.
+
 It's generally best to pick the most restricted bot type that is sufficient
 to do the task at hand. Anyone with the bot's API key can do anything the
 bot can.
```
OK, but the "we" is anti-pattern, so:

> Use the most limited bot type that supports your integration. Anyone with the
bot's API key can do anything the bot can, so giving bots unnecessary
permissions can expose your organization to unnecessary risk.

alya commented on starlight_help/src/content/docs/bots-overview.mdx:
```diff
@@ -42,31 +42,32 @@ The **bot type** determines what the bot can do.
+
+Bot can be subscribed to channels, and their [channel
+permissions](/help/channel-permissions) are the same for bots as for other
+users. For example, for private channels with protected history, a bot can only
```
How about:

> Bots can be subscribed to channels, and assigned [channel
permissions](/help/channel-permissions) just like human users. In private
channels with protected history, a bot can only access messages sent after it
was subscribed to the channel.

alya commented:
I pushed changes addressing the feedback above, but the new `include` file isn't working, and I don't understand why. [#documentation > confused about include file](https://chat.zulip.org/#narrow/channel/19-documentation/topic/confused.20about.20include.20file/with/2261304)

timabbott commented:
I can fix the merge conflicts with the "Help Center" => "help center" PR and merge once I'm done with another task.

timabbott commented:
Resolved the new merge conflicts and merged, thanks @alya!
